t/perf/opcount.t

```diff
@@ -796,4 +796,34 @@ test_opcount(0, "sassign + padsv replaced by padsv_store",
                     padsv_store  => 2,
                 });
 
+# OPpTARGET_MY optimizations on undef
+test_opcount(0, "undef + padsv (undef my \$x) is reduced to undef",
+                sub { undef my $x },
+                {
+                    undef   => 1,
+                    padsv   => 0,
+                    sassign => 0,
+                });
+test_opcount(0, "undef + padsv + sassign (my \$x = undef) is reduced to undef",
```

Comment:
I think id expect to see some aassign tests as well. 

```
my $y= 1; my @x= ($y= undef);
my $x= 1; sub f{} f($x=undef);
my ($x,$p)=(1,2); sub f{} f(($x=undef),$p);
```
also maybe these too (not aassign related).
```
my $h= {}; my @k= keys %{($h=undef)||{}};
my $y= 1; my @x= \($y= undef);
```

Reply:
Tests in t/perf/opcount.t or as general tests?

Reply:
It's okay, I'll add them in as general tests as well.

Reply:
I've added the tests to t/op/undef.t as well. Wasn't sure exactly what you wanted to test (apart from serious breakage), so feel free to give them a quick look over.